src/templates/plugin.main.js

```diff
@@ -90,10 +90,22 @@ export default async (context) => {
     // Lazy-loading enabled
     if (lazy) {
       const { loadLanguageAsync } = require('./utils')
-
-      // Load fallback locale.
-      if (app.i18n.fallbackLocale && newLocale !== app.i18n.fallbackLocale) {
-        await loadLanguageAsync(context, app.i18n.fallbackLocale)
+      const i18nFallbackLocale = app.i18n.fallbackLocale
+
+      // Load fallback locale(s).
+      if (i18nFallbackLocale) {
+        let localesToLoadPromises = []
+        if (Array.isArray(i18nFallbackLocale)) {
+          localesToLoadPromises = i18nFallbackLocale.map(fbLocale => loadLanguageAsync(context, fbLocale))
+        } else if (typeof i18nFallbackLocale === 'object' && i18nFallbackLocale[newLocale]) {
```

Comment:
This logic seems wrong?

If using the object notation and there is no explicit fallback for `newLocale` then we should check `i18nFallbackLocale.default`, not fallback to "string" notation below.

Reply:
Not sure to understand. If we are in an object notation, and there is no default defined, then nothing more happens. The following else is related to the condition tree (array, object, string), no to the `if` about the default.

Reply:
The code is:

```js
        } else if (typeof i18nFallbackLocale === 'object' && i18nFallbackLocale[newLocale]) {
          // ...
        } else if (newLocale !== i18nFallbackLocale) {
          localesToLoadPromises.push(loadLanguageAsync(context, i18nFallbackLocale))
        }
```

So if `i18nFallbackLocale[newLocale]` is falsey, it will go to the next `if` condition which will be true because `newLocale !== i18nFallbackLocale` and will pass the whole fallback object to `loadLanguageAsync`.

Reply:
You're right. I've messed up the condition. I've changed it.

I have refactored the tests too, and added another one. Don't hesitate to tell be anything wrong.